File: Exemplos/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/SAT/ExtrairLogs.ts
import {SatCommand} from './SatCommand';

export class ExtrairLogs extends SatCommand {
  readonly numSessao: number;
  readonly codAtivacao: string;

  constructor(numSessao: number, codAtivacao: string) {
    super('ExtrairLogs');
    this.numSessao = numSessao;
    this.codAtivacao = codAtivacao;
  }

  functionParameters(): object {
    return {numSessao: this.numSessao, codAtivacao: this.codAtivacao};
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid/IntentServices/Sat/ExtrairLogs.cs
namespace Xamarin_Android_Intent_Digital_Hub.IntentServices.Sat
{
    class ExtrairLogs : SatCommand
    {
        readonly int numSessao;
        readonly string codAtivacao;

        public ExtrairLogs(int numSessao, string codAtivacao) : base("ExtrairLogs")
        {
            this.numSessao = numSessao;
            this.codAtivacao = codAtivacao;
        }

        protected override string FunctionParameters()
        {
            return "\"numSessao\"" + ":" + numSessao + "," +
                    "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"";
        }
    }
}